src/model/entities/rundown.ts

```diff
@@ -118,9 +116,12 @@ export class Rundown extends BasicRundown {
   }
 
   private findFirstSegment(): Segment {
-    return this.segments.reduce((previousSegment: Segment, currentSegment: Segment) => {
-      return previousSegment.rank < currentSegment.rank ? previousSegment : currentSegment
-    })
+    return this.segments
+      .filter(segment => this.isSegmentValidForRundownExecution(segment))[0]
```

Comment:
This can be rewritten as
```suggestion
    return this.segments
      .find(segment => this.isSegmentValidForRundownExecution(segment))
```

And note that the return type of the method declaration and the implementation are not aligning. The implementation returns `Segment | undefined`.

Reply:
Updated to use `.find()`.
The return type is still just `Segment`, however, the method throws an exception if it couldn't find a Segment.